### DISCUSSION TITLE
Contributions / sponsorship?

### DISCUSSION THREAD

**Discussion by shawaj** [librespot-org/librespot#613] [General]
It would be nice to have a way to contribute to this project. Is there any plans to set up a funding page / sponsorship? Or would that cause issues with Spotify then considering it to be commercial?

Perhaps something like open collective could be good - for example - https://opencollective.com/reviewdog

**Comment by lufinkey** [comment]
I second this. I would love to give money to keep this project updated

**Comment by sashahilton00** [comment]
It's great to hear that you'd like to support the project. Whilst I can't speak for everyone, The reason I personally haven't looked to add sponsorship to this project is primarily due to the fact that money is not so much the issue as time; myself and the other maintainers are all, I believe, pretty fully occupied, and thus this project can only ever be a hobby project currently. Whilst learning rust properly is something that I would like to do at some point, the fact of the matter is that I haven't really had any time to tend to the issues on this repo reliably, much less learn the language to a degree that is sufficient to rearchitect the application to solve some of the core issues such as session reconnection. There is then the secondary problem of having to reverse engineer the Spotify app, itself a time consuming process, not helped by the fact that their infrastructure appears to be a chaotic mess that lacks any sort of stable APIS outside of the publicly documented ones, and that they go out of their way to make it difficult to use their private ones.

If there were an experienced rust developer that wanted to spend time maintaining this project, and a contribution/sponsorship option helped that, I'd be all in favour of adding it. But as it stands, it would feel like taking money for not doing very much, whilst creating other issues such as how one distributed funds in a fair manner. What might work is some sort of Bountysource solution for important features/bugs, I'd be interested to hear from anyone who has thoughts on that.

**Comment by lufinkey** [comment]
I would be fine with doing bug bounties or something like that. I very much want/need this library to continue being updated

**Comment by sashahilton00** [comment]
Have had a look at bountysource, and on second thoughts their terms and conditions don't look so great. Open Collective doesn't seem like a bad idea in that it could potentially provide a bit of a slush fund to pay people to work on more challenging features/bugs that would otherwise go unattended. Would that be of interest?
